packages/connection/legacy/class-jetpack-signature.php

```diff
@@ -361,17 +365,40 @@ public function get_current_request_port() {
 			// if the site is behind a proxy running on port 443 without
 			// X-Forwarded-Port and the back end's port is *not* 80. It's better,
 			// though, to configure the proxy to send X-Forwarded-Port.
-			$https_port = defined( 'JETPACK_SIGNATURE__HTTPS_PORT' ) ? JETPACK_SIGNATURE__HTTPS_PORT : 443;
-			$port       = in_array( $host_port, array( 443, 80, $https_port ), false ) ? '' : $host_port; // phpcs:ignore WordPress.PHP.StrictInArray.FoundNonStrictFalse
+			$https_port = defined( 'JETPACK_SIGNATURE__HTTPS_PORT' ) ? $this->sanitize_host_post( JETPACK_SIGNATURE__HTTPS_PORT ) : '443';
+			$port       = in_array( $host_port, array( '443', '80', $https_port ), true ) ? '' : $host_port;
 		} else {
 			// 80: Standard Port
 			// JETPACK_SIGNATURE__HTTPS_PORT: Set this constant in wp-config.php to the back end webserver's port
 			// if the site is behind a proxy running on port 80 without
 			// X-Forwarded-Port. It's better, though, to configure the proxy to
 			// send X-Forwarded-Port.
-			$http_port = defined( 'JETPACK_SIGNATURE__HTTP_PORT' ) ? JETPACK_SIGNATURE__HTTP_PORT : 80;
-			$port      = in_array( $host_port, array( 80, $http_port ), false ) ? '' : $host_port; // phpcs:ignore WordPress.PHP.StrictInArray.FoundNonStrictFalse
+			$http_port = defined( 'JETPACK_SIGNATURE__HTTP_PORT' ) ? $this->sanitize_host_post( JETPACK_SIGNATURE__HTTP_PORT ) : '80';
+			$port      = in_array( $host_port, array( '80', $http_port ), true ) ? '' : $host_port;
 		}
 		return (string) $port;
 	}
+
+	/**
+	 * Sanitizes a variable checking if it's a valid port number, which can be an integer or a numeric string
+	 *
+	 * @since 9.2.0
+	 *
+	 * @param mixed $port_number Variable representing a port number.
+	 * @return string Always a string with a valid port number, or an empty string if input is invalid
+	 */
+	public function sanitize_host_post( $port_number ) {
+
+		if ( ! is_int( $port_number ) && ! is_string( $port_number ) ) {
+			return '';
+		}
+		if ( is_string( $port_number ) && ! ctype_digit( $port_number ) ) {
+			return '';
+		}
+
+		if ( 0 >= (int) $port_number ) {
```

Comment:
Should you also check that the port isn't above 65535?

Reply:
good call. 

I tested the constants in many ways, but these have a big impact on the time taken to run tests. We have to consider when this is going to be a problem

Reply:
PR updated with the port range validation

Reply:
If we test the sanitization function that's used for the constants (which you are), there shouldn't be a need to test all failure cases for the constants themselves.

We could also look at where the overhead is actually coming from, and see if we can reduce it. Maybe your idea of not initializing WorDBless unnecessarily in bootstrap.php would be good for that purpose after all. But perhaps that can wait for a followup PR.

Reply:
that's what I was thinking. The method uses one WordPress function (`is_ssl`) that could be easily mocked.

But I'm not sure the impact would be too big. It seems to me that what takes time is to setup happens before we get to `bootstrap`, so not loading the WordPress files wouldn't change the time that much. But I can give it a try.

And I agree that only one test for the constants should be enough

Reply:
Confirmed that the time difference if we don't load WordPress files is irrelevant.

Removed 2 redundant constants tests